generators/typescript-v2/ast/src/ast/FunctionInvocation.ts

```diff
@@ -0,0 +1,27 @@
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+import { Func } from "./Func";
+import { Reference } from "./Reference";
+
+export declare namespace FunctionInvocation {
+    interface Args {
+        func: Reference;
+        arguments_: AstNode[];
+    }
+}
+
+export class FunctionInvocation extends AstNode {
+    private func: Reference;
+    private arguments_: AstNode[];
+
+    constructor({ func, arguments_ }: FunctionInvocation.Args) {
+        super();
+        this.func = func;
+        this.arguments_ = arguments_;
+    }
+
+    public write(writer: Writer): void {
+        writer.writeNode(this.func);
+        writer.delimit(this.arguments_, ", ", (argument) => argument.write(writer), "(", ")");
```

Comment:
Thoughts on using an approach like `writeArguments` like I mentioned [here](https://github.com/fern-api/fern/pull/5422#discussion_r1887542602)? Otherwise we would need to refactor this to handle the `nop` nodes again separately.